jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -588,6 +588,10 @@ ecma_builtin_typedarray_prototype_filter (ecma_value_t this_arg, /**< this argum
 
     ret_value = ecma_op_create_typedarray_with_type_and_length (obj_p, pass_num);
 
+    if (len == 0)
+    {
+      return ret_value;
+    }
```

Comment:
Hm, why this is not before the define local array?

Reply:
Because in some cases we can not return with `ECMA_VALUE_EMPTY`

Reply:
so we return with a new typedarray?

Reply:
actually we return from a macro enclosed block, and that is not necessarily healthy.

Reply:
Yes but we only return from there when the length is 0, therefore the enclosing macro would basically do nothing

Reply:
And yes, we have to give back a 0 length typedarray in that case

Reply:
The problem is we might change silently the macro and we have a hidden bug. Creating and returning with an empty array early should be easy.

Reply:
Yes, that is definitely right, it would just mean a bit if a code duplication there.

Reply:
The compiler should handle that. Since it is a single line, I would not even call it a duplication.

Reply:
Yeah you convinced me there, I'm updating the PR